src/plugins/data_source/common/data_sources/types.ts

```diff
@@ -10,6 +10,7 @@ export interface DataSourceAttributes extends SavedObjectAttributes {
   description?: string;
   endpoint: string;
   dataSourceVersion?: string;
+  dataSourceEngineType?: DataSourceEngineType;
```

Comment:
do we need a migration? i dont think so right 

https://github.com/opensearch-project/OpenSearch-Dashboards/blob/a4aa682226b9d22d2c6e7a41a11a44225b6d7db1/src/plugins/data_source/server/saved_objects/data_source.ts#L44

it's optional field. so think it doesn't need it.

Reply:
that's why this field is marked optional yes